Core/Versioning/KspVersion.cs

```diff
@@ -444,6 +444,53 @@ public static bool TryParse(string input, out KspVersion result)
                 return false;
             }
         }
+
+        /// <summary>
+        /// Searches the build map if it knows the version.
+        /// </summary>
+        /// <returns><c>true</c>, if version is valid, <c>false</c> otherwise.</returns>
+        public bool IsValid()
+        {
+            List<KspVersion> knownVersions = new KspBuildMap(new Win32Registry()).KnownVersions;
+
+            foreach (KspVersion ver in knownVersions)
+            {
+                if (ver.Major == Major && ver.Minor == Minor && ver.Patch == Patch)
+                {
+                    if (ver.Build == Build || !IsBuildDefined)
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Returns a complete KspVersion object, including the build number.
+        /// If a version number has multiple possible builds, it takes the earliest one.
+        /// </summary>
+        /// <returns>The build for the version. Null if version is not known in the build map.</returns>
+        public KspVersion AddBuildToVersion ()
+        {
+            if (!IsPatchDefined)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+
+            List<KspVersion> knownVersions = new KspBuildMap(new Win32Registry()).KnownVersions;
```

Comment:
So add a 'knownVersion.Reverse()' here to in-place reverse the list, and we'll find the last valid build number, most likely faster.

Reply:
Yep, good idea 👍